source/MaterialXView/Handlers/DefaultGeometryHandler.h

```diff
@@ -0,0 +1,51 @@
+#ifndef MATERIALX_DEFAULTGEOMETRYHANDLER_H
+#define MATERIALX_DEFAULTGEOMETRYHANDLER_H
+
+#include <string>
+#include <memory>
+#include <MaterialXView/Handlers/GeometryHandler.h>
+
+namespace MaterialX
+{
+/// Shared pointer to an GeometryHandler
+using DefaultGeometryHandlerPtr = std::shared_ptr<class DefaultGeometryHandler>;
+
+/// @class @GeometryHandler
+/// Utility geometry handler for providing data for shader binding.
+///
+class DefaultGeometryHandler : public GeometryHandler
+{
+  public:
+    /// Static instance creator
+    static DefaultGeometryHandlerPtr creator() { return std::make_shared<DefaultGeometryHandler>(); }
+
+    /// Default constructor
+    DefaultGeometryHandler();
+    
+    /// Default destructor
+    virtual ~DefaultGeometryHandler();
+
+    void setIdentifier(const std::string identifier) override;
+
+    unsigned int* getIndexing(size_t &bufferSize, unsigned int& indexCount) override;
+    float* getPositions(size_t &bufferSize, unsigned screenWidth = 0, unsigned int screenHeight = 0) override;
+    float* getNormals(size_t &bufferSize) override;
+    float* getTextureCoords(const std::string& setIdentifier, size_t &bufferSize) override;
+    float* getTangents(const std::string& setIdentifier, size_t &bufferSize) override;
+    float* getBitangents(const std::string& setIdentifier, size_t &bufferSize) override;
+    float* getColors(const std::string& setIdentifier, size_t &bufferSize) override;
```

Comment:
Maybe change the name of input variable "setIdentifier"? It sounds like a method name.. And in fact there is a method called that just above :D

I'm surprised the compiler didn't say anything about that. What if you called the method setIdentifier with the input variable setIdentifier? setIdentifier(setIdentifier) Would that be allowed? :)

Reply:
attributeIdentifier ?

Reply:
Maybe this should be an integer index even, following the MaterialX convention? I think that would be better. Just call it index in that case, and describe in the docstring what it's for.